src/test/kotlin/com/atlassian/performance/tools/ssh/api/SshTest.kt

```diff
@@ -56,7 +56,8 @@ class SshTest {
         SshContainer().useSsh { sshHost ->
             installPing(sshHost)
 
-            val fail = sshHost.runInBackground("nonexistent-command")
+            val fail = sshHost.runInBackground("nonexistant-command")
+            sshHost.waitForAllProcessesToFinish("nonexistant-command", Duration.ofMillis(100))
```

Comment:
Now the test shows that the `runInBackground` API cannot be used easily. We should fix the problem in the API rather than in tests.

Reply:
Another approach would be to just return some specific `SshConnection.SshResult` as part of `BackgroundProcess.stop` when we don't really get the `exitStatus`, however I don't know what would that be and we need to return something if we want to maintain the current API of `BackgroundProcess`

Reply:
@dagguh I'd like to fix the flakiness of the test. This is my main goal.

My understanding of possible implementations of your suggestion is to either make it less portable or break the API (see my 2 previous comments). I don't like any of those options and I'm a bit stuck with this PR. Do you have any other ideas how this could be fixed? If not then maybe you have opinion about which of those 2 is better?

Reply:
@pczuj why are we afraid to break the API and properly apply semantic versioning?

Reply:
>  break the API and properly apply semantic versioning

Yes, we can just break the API, it's a tool in our belt.
It can require some extra effort. Let's say we release `ssh:3.0.0`. The`infrastructure` is a `ssh` consumer:
```
api("com.atlassian.performance.tools:ssh:[2.3.0,3.0.0)")
```
It will have a choice:

1. `ssh:[2.3.0,3.0.0)` - don't use the new `ssh`, this is the default state. It takes no effort, but what was the point of `ssh:3.0.0`, if no consumer needs it?
2. `ssh:[2.3.0,4.0.0)` - use the new, without dropping the old. Only possible if the removed API was not used.
3. `ssh:[3.0.0,4.0.0)`- use the new, drop the old. This means that all consumers of `infrastructure`, which are still on `ssh:[2.0.0,3.0.0)`, will no longer be compatible. Therefore that bump is breakage of `infrastructure` (its POM contract), which is a another major release. This can cascade to multiple libs: `aws-infrastructure` and `jira-performance-tests`.

We used scenario 3 many times in the past. It's a bit annoying, but not that scary.
We successfully [used scenario 2](api("com.atlassian.performance.tools:ssh:[2.3.0,3.0.0)")) as well.

Reply:
> We should fix the problem in the API rather than in tests.

I didn't mean we have to break API. I meant: the tests found a real flakiness pain, let's fix it for the consumers as well.

>  this approach is very system specific (usage of `wait` and `pgrep`)

I would avoid it too. Not only does it lose generality, but also more moving parts: brittle and complex.